Hw05v2

StanislavAntunovich opened the pull request:
Переведите ваше приложение на Фрагменты по аналогии с CityInfo.

StanislavAntunovich committed:
К этому моменту у вас уже должно быть как минимум два готовых экрана: экран выбора города и экран отображения температуры для выбранного города. Свяжите два экрана между собой через интент - передавайте данные, которые ввел пользователь (название города, дополнительные параметры), на экран отображения температуры и показывайте их там (температура и значения дополнительных параметров могут быть вымышленными)
* Добавьте отдельную кнопку на экране отображения температуры, которая будет открывать браузер и показывать какую-то дополнительную информацию (например, статью о городе из википедии, прогноз погоды на неделю из Яндекса или чем знаменателен этот день).

StanislavAntunovich committed:
adding fragments and data classes

StanislavAntunovich committed:
adapter

StanislavAntunovich committed:
adapter

StanislavAntunovich committed:
recycler found

StanislavAntunovich committed:
changing weather details on swiping city-cards

StanislavAntunovich committed:
Переведите ваше приложение на Фрагменты по аналогии с CityInfo.

DwarfPorter commented on app/src/main/java/ru/geekbrains/android1/WeatherSettingsParcel.java:
```diff
@@ -1,29 +1,14 @@
+import java.io.Serializable;
 
-    private static final Object lock = new Object();
+public class WeatherSettingsParcel implements Serializable {
```
не нашел, где этот класс используется

DwarfPorter commented on app/src/main/java/ru/geekbrains/android1/CityWeatherAdapter.java:
```diff
@@ -0,0 +1,65 @@
+
+    public CityWeatherAdapter(String[] cities, String[] conditions, MainWeatherFragment parent) {
+        this.cities = cities;
+        this.data = FakeData.getInstance();
```
а зачем мы данные делаем синглтоном?
Этот подход неверный. Данных может быть много, они должны быть изменяемыми, и сборщик мусора должен их чистить время от времени.

StanislavAntunovich committed:
menu added
activity with adding/removing cities added (recycler view)
refactoring (
    less singletons more parcels
    init views and parts moved out from onCreate
    )

StanislavAntunovich committed:
back button pressed from add city crash fixed

StanislavAntunovich committed:
migration on androidx

StanislavAntunovich committed:
Merge pull request #6 from StanislavAntunovich/hw06

hw06
